staging/src/k8s.io/api/storage/v1beta1/types.go

```diff
@@ -209,3 +209,203 @@ type VolumeError struct {
 	// +optional
 	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CSIDriver captures information about a Container Storage Interface (CSI)
+// volume driver deployed on the cluster.
+// CSI drivers do not need to create the CSIDriver object directly. Instead they may use the
+// cluster-driver-registrar sidecar container. When deployed with a CSI driver it automatically
+// creates a CSIDriver object representing the driver.
+// Kubernetes attach detach controller uses this object to determine whether attach is required.
+// Kubelet uses this object to determine whether pod information needs to be passed on mount.
+// CSIDriver objects are non-namespaced.
+type CSIDriver struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object metadata.
+	// metadata.Name indicates the name of the CSI driver that this object
+	// refers to; it MUST be the same name returned by the CSI GetPluginName()
+	// call for that driver.
+	// The driver name must be 63 characters or less, beginning and ending with
```

Comment:
The CSI spec references [rfc1035](https://tools.ietf.org/html/rfc1035#section-2.3.1) for valid syntax. Can we use `IsDNS1035Label` to validate the csi name instead?

Reply:
I can do that, but it will give us different results from apivalidation.ValidateCSIDriverName as "_" won't be supported.  I can add a different validation.

Reply:
The csi spec does not allow "_", so it appears that `ValidateCSIDriverName` is wrong. Can we fix the original validation method? It would technically be a breaking change since we would be tightening validation :-/

Reply:
Will we be breaking any drivers by doing this?  I wonder if we need to add warming messages first and then remove support for _?

Reply:
What is the difference between DNS1035 and csi driver name?

@saad-ali and I checked all the drivers listed [here](https://kubernetes-csi.github.io/docs/drivers.html), and none of them have underscores.  It's still not completely safe to tighten validation because folks could have written private csi drivers not intended for public consumption.  

Saad suggested that we just send out an email and release notes announcement that we're going to start enforcing the csi spec driver name in Kubernetes 1.14, since the requirement has already been documented in the CSI spec.  Does this sound like a reasonable exception to the tightening validation rule @thockin?

Reply:
Yes.